### PR TITLE
hooks-api docs

### PR DESCRIPTION
**Pull request by cviejo**

### CODE REVIEW

File: hooks-api.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# in-context hook api
+
+hooks are a means to attach jobs at specific moments of a task's lifecycle.
```
Comment (JohannesMerz) [inline hooks-api.md]: means sounds off for me, i would use `way` instead

File: hooks-api.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# in-context hook api
+
+hooks are a means to attach jobs at specific moments of a task's lifecycle. 
+
+The jobs run on the same context as the displayed content, typically that would be the top frame. They are also blocking: if a promise is returned, the internal job runner will wait for it to resolve before advancing to he next task stage
```
Comment (JohannesMerz) [inline hooks-api.md]:
```suggestion
The jobs run on the same context as the displayed content, typically that would be the top frame. They are also blocking: if a promise is returned, the internal job runner will wait for it to resolve before advancing to he next task stage.
```

File: hooks-api.md (hunk 1 above)
Comment (JohannesMerz) [inline hooks-api.md]:
```suggestion
Hooks are a way to attach jobs at specific moments of a task's lifecycle. 
```

File: hooks-api.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+Two types of hook can be set using:
+* `setInitHook(fn)`: job will run only once, the first time the script is used, before the task begin.
```
Comment (JohannesMerz) [inline hooks-api.md]:
```suggestion
* `setInitHook(fn)`: job will run only once, the first time the script is used, before the task begins.
```

File: hooks-api.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+* `setTaskHook(stage, fn)`: job runs on each task.
+
+This is a sample application that loads an external sdk and calls its methods. For brevity, we assume two functions `createScript` and `waitFor` that load a script and wait until a condition is fulfilled:
+
+```javascript
+window.setInitHook(() => {
```
Comment (JohannesMerz) [inline hooks-api.md]:
```suggestion
window.setInitHook(async () => { 
```

File: hooks-api.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  In a two task setup, the script above should print:
+  ```log
+  task count: 1
+  task count: 2
+  ```
+
+- `render` is a utility function for rendering custom content. It's meant as a way to help hide and manage the rendered content after each stage
+  ```javascript
+  window.setTaskHook("preparation", ({ render }) => ({
+    const warningDiv = document.createElement('div');
```
Comment (JohannesMerz) [inline hooks-api.md]: 💭  maybe a more thorough example in here would be good to also explain the blocking character of jobs when return a promise as thats missing in the whole doc so far.

**Review by JohannesMerz — CHANGES_REQUESTED** [review]

File: hooks-api.md (hunk 1 above)
Comment (cviejo) [reply hooks-api.md]: https://www.linguee.com/english-german/search?source=auto&query=is+a+means+to

it's correct english, though. `a means to`, `a means of`, `a means to an end`

File: hooks-api.md (hunk 5 above)
Comment (cviejo) [reply hooks-api.md]: i was doing that while you were reviewing 😃

**Review by JohannesMerz — APPROVED** [review]

**Pull request merged**